rest/route/host_agent_test.go

```diff
@@ -994,6 +995,65 @@ func TestAssignNextAvailableTaskWithDispatcherSettingsVersionLegacy(t *testing.T
 			So(shouldTeardown, ShouldBeFalse)
 			So(t, ShouldBeNil)
 		})
+		Convey("a task's NumNextTaskDispatches should be increased by 1", func() {
```

Comment:
Since the above logic was modified to increment the counter when sending back an already-assigned task, should there be a test for when the task reaches the dispatching limit? Would be good to know that the route fixes the host/task as expected when it hits the limit.